assets/src/edit-story/components/form/advancedDropDown/list/list.js

```diff
@@ -0,0 +1,281 @@
+      } else if (key === 'ArrowDown') {
+        setFocusIndex((index) =>
+          Math.min(filteredOptions.length - 1, index + 1)
+        );
+      }
+    },
+    [focusIndex, filteredOptions, onClose, onSelect]
+  );
+
+  useFocusOut(listRef, () => setFocusIndex(-1), []);
+
+  // scroll into view and focus focusIndex
+  useEffect(() => {
+    const listEl = listRef.current;
+    if (!listEl) {
+      return;
+    }
+    if (focusIndex === -1) {
+      listEl.scrollTo(0, 0);
+      return;
+    }
+
+    const highlighedOptionEl = optionsRef.current[focusIndex];
+    if (!highlighedOptionEl) {
+      return;
+    }
+
+    highlighedOptionEl.focus();
+    listEl.scrollTo(0, highlighedOptionEl.offsetTop - listEl.clientHeight / 2);
+  }, [focusIndex, filteredOptions, keyword]);
+
+  /*
+   * ACCESSIBILITY COMMUNICATION WITH SEARCH INPUT
+   */
+  const isExpanded = filteredOptions.length > 0;
+  useEffect(() => onExpandedChange(isExpanded), [onExpandedChange, isExpanded]);
+  useEffect(() => {
+    if (focusTrigger > 0) {
+      setFocusIndex(0);
+    }
+  }, [focusTrigger]);
+
+  return filteredOptions.length <= 0 ? (
+    <NoResult>{__('No matches found', 'web-stories')}</NoResult>
+  ) : (
+    <List
+      ref={listRef}
+      tabIndex={0}
+      id={listId}
+      role="listbox"
+      onKeyDown={handleKeyPress}
+      aria-label={__('Option List Selector', 'web-stories')}
+      aria-required={false}
+    >
+      {filteredListGroups.map((group, i) => {
+        const groupLabelId = `group-${uuidv4()}`;
+        return (
+          group.options.length > 0 && (
+            <Group
+              key={group.label}
+              role="group"
+              aria-labelledby={groupLabelId}
+            >
+              {group.label && (
+                <GroupLabel id={groupLabelId} role="presentation">
+                  {group.label}
+                </GroupLabel>
+              )}
+              {group.options.map((option, j) => {
+                const optionProps = {
+                  key: option.id,
+                  role: 'option',
+                  tabIndex: -1,
+                  'aria-selected': value === option.id,
+                  'aria-posinset': getInset(filteredListGroups, i, j),
+                  'aria-setsize': filteredOptions.length,
+                  'data-option': option.id,
+                  onClick: () => onSelect(option.id),
+                  ref: (el) =>
+                    (optionsRef.current[
+                      getInset(filteredListGroups, i, j)
+                    ] = el),
+                };
+                return renderer ? (
+                  <OptionRenderer {...optionProps} option={option} />
+                ) : (
+                  <Option {...optionProps}>
+                    {value === option.id && (
+                      <Selected aria-label={__('Selected', 'web-stories')} />
+                    )}
+                    {option.name}
+                  </Option>
+                );
```

Comment:
Could this be done with a default value instead? So it's defined in the component as `function OptionList({ ..., renderer = DefaultRenderer, ... })` - then you can also inline all the props rather than define them in a props-object.